modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/TeamMembersTable/components/Filter/index.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import {useEffect, useRef} from 'react';
+
+import SearchBar from '../../../../components/SearchBar';
+
+const TeamMembersFilter = ({userAccounts, filtersState: [setFilters]}) => {
+	const countFetchUserAccountsRef = useRef(0);
+
+	useEffect(() => {
+		if (userAccounts) {
+			countFetchUserAccountsRef.current = ++countFetchUserAccountsRef.current;
+		}
+	}, [userAccounts]);
+
+	return (
+		<div className="d-flex flex-column">
+			<div className="d-flex">
+				<SearchBar setFilters={setFilters} />
+			</div>
+		</div>
+	);
+};
+
+export default TeamMembersFilter;
```

Comment:
@jairmedeiros can you explain more about the need about this useRef?

Reply:
@igordmf

Reply:
useRef return a object that run "outside" from React (virtual DOM), by DOM (https://reactjs.org/docs/faq-internals.html). In this way, it is possible to update your value and the React doesn't update/re-rendered. We use it as a workaround to update  the countFetchUserAccountsRef to avoid looping

Reply:
https://www.youtube.com/watch?v=t2ypzz6gJm0. This video may be helpful

Reply:
I think Lucas is asking about this specific useRef, seems like it isn't beeing used

Reply:
yea, my bad. As I saw 3 approves and this questions I supposed that was about the concept.